### ISSUE TITLE
QUERY_TIMEOUT: Your query request was running for too long

### ISSUE THREAD

**Issue by etrabelsi** [superfell/Beatbox#45]
Can you add support for setting timeout in beatbox as parameter to the constructor ?

**Comment by hynekcer** [comment]
I agree. Which one source do you use? Does this work for also you? (still without timeout setting?)

  pip install git+https://github.com/hynekcer/beatbox-davisagli.git@7f628a789cba#egg=beatbox

If so then I can add the timeout setting and ask for commit to PyPI, otherwise the project "hibernation" would continue because there are different clones by more authors

**Comment by etrabelsi** [comment]
It seems to be weird fork of this repository ,it doesn't support this kind of login.

> self.sf = beatbox._tPartnerNS
> self.svc = beatbox.Client()
> self.svc.login(user_name, password + security_token)

**Comment by superfell** [comment]
Do you want to set the timeout in the hopes of extending it so as to not get a QUERY_TIMEOUT result? if so that won't help, the QUERY_TIMEOUT response is generated from a server side timeout, and is not adjustable from the client.

**Comment by hynekcer** [comment]
I think that a frequent requirement is to set a shorter value for the case that a simple query is unusually slow or the connection is sometimes broken but not known to be closed. Especially on a web server is a shorter timeout important because the user will more probably retry than wait. It usually works good surprisingly because the cache on SFDC is filled by necessary data by the first terminated request and the second request frequently succeeds fast.
